core/metadata-derive/src/encode.rs

```diff
@@ -0,0 +1,206 @@
+use quote::{quote, quote_spanned};
```

Comment:
I have no experience writing procedural macros.  @rphmeier can you help review this?

Reply:
I don't have much experience with proc-macros or this code in particular either.